groupsforsections.py

```diff
@@ -255,7 +283,39 @@ def main():
 
         # mapping all the users in a sections to corresponding group
         groups_to_users_dict[group_id] = users
-        # TODO: Adding users to corresponding groups
+
+    failed_groups_to_users_dict={}
+    success_groups_to_users_dict={}
+
+    # adding users to the group
+    for group in groups_to_users_dict:
+        for user in groups_to_users_dict[group]:
+            membership_id = add_users_to_group(groups_for_section_class, group, user)
+            if membership_id is None:
+                logging.error('The user %s is not added to the group %s' % (user, group))
+                failed_groups_to_users_dict.setdefault(group, []).append(user)
+            else:
+                success_groups_to_users_dict.setdefault(group, []).append(user)
```

Comment:
I've always found the use of `setdefault()` like this to be strange.  Recent versions of Python (2.5) added a subclass of `dict` called `defaultdict`.  When a dictionary is instantiated, new keys in the dictionary will be initialized with the first value passed to the class.  Using it in this case (and combining code from my earlier comment):

```python
from collections import defaultdict
#...
    failed_groups_to_users_dict=defaultdict(list)
    success_groups_to_users_dict=defaultdict(list)

    for group, user in groups_to_users_dict.items():
        membership_id = add_users_to_group(groups_for_section_class, group, user)
        if membership_id is None:
            logging.error('The user %s is not added to the group %s' % (user, group))
            failed_groups_to_users_dict[group].append(user)
        else:
            logging.info('user "%s" added to group "%s" with ID (%d)' %(user, group, membership_id))
            success_groups_to_users_dict[group].append(user)